### DISCUSSION TITLE
at the end of installation having issues.

### DISCUSSION THREAD

**Discussion by ineptmath** [alexhorner/WD-DX4000-Installer#9] [Q&A]
hey people, at the end of the installation I'm having issues do this 
```text
Mount your installation's boot partition (usually the very first partition on the newly installed disk) and copy startup.nsh from the root of the installer USB to the root of your boot partition. You may now reboot and the system should come online on it's own. 
```
I've tried reading to find the answers I'm just not that adept in Linux i guess. ill remove this or add an answer if i figure it out before i get a response thanks.

id add more verbose information but the commands i wont to use don't even seem to work in this shell.

**Comment by alexhorner** [comment]
Hi there,

You won't find answers specific to startup.nsh copying elsewhere at all, as it is unique to this device.

This installation (and subsequent usage) lands you with a clean Debian install without any kind of GUI, so there is to an extent an expectation of experience using Debian as a prerequisite to this install.

Please could you provide the commands you have attempted to use, and screenshots of where you have tried to use them?

Thanks

**Comment by ineptmath** [comment]
![image](https://github.com/user-attachments/assets/f8b06616-bc67-4220-b9ae-f9b4bb836197)
i tryed to copy this 
![image](https://github.com/user-attachments/assets/861f1191-29d7-40de-8ed5-5f104738ed8d)
but i guess im missing some steps. i guess i realy just dont know the commands im meant to be typing to show me the filesystem's partitioning.
install was just generic  default options, except no gui/wm (yes ironic i know but i will get there). i dunno if i was meant to save any of the drive information (first time going down this path). fwiw 4 2tb matching drives in stock 'recreate' raid5. hope that helps thanks

**Reply by alexhorner** [reply]
Hi there,

This is intended to be used without any of the stock RAID configuration. This system uses Intel Rapid RAID which in my experience greatly confuses Linux when it tries to work out the drives. It is possible to import an existing array if you'd like to keep data, but that doesn't seem to apply to your scenario here. After installation I would recommend ZFS or MD.

I would suggest you choose one drive to install to, or attach an external HDD or SSD (USB drives can wear out quickly) and install to that.

You have the right idea with the commands. You want to `disk-detect` to populate the nodes under `/dev` and then you want to find the `/dev/sdX` drive matching the drive you installed to (where X is the letter of the drive). You can list block devices and their size with `lsblk`. Mount the partition as described in the guide: Make a mountpoint such as `mkdir /mnt/bootpart` and then mount with `mount /dev/sdXY /mnt/bootpart`, where X is the letter and Y is the partition number (usually the first partition is the boot partition). You can see with `df` whether it is mounted or not afterwards. Verify the partition is the right one (look at the contents of `/mnt/bootpart`) and then copy the `startup.nsh` from the installer over to it and reboot.

**Reply by ineptmath** [reply]
thanks for your support, i dont seem to have ```lsblk``` command working i will try again later 
im unable to mount any of the sdx drives apart from the media which i presume is the USB(sda2)

ok i boot to usb, ```ls dev/``` provides me with  ```sda, sda1, sda2```
i then run ```disk-detect``` which then provides ``` ls dev/ with sdb and sdc``` 
i then used ```e2fsck -f /dev/sdX```
which provides me with this
![image](https://github.com/user-attachments/assets/03a46a49-3a0f-4042-ad1a-5dbcf0cb0bc7)
![image](https://github.com/user-attachments/assets/894b7ae2-b535-4727-b977-0bb742dc419e)

sorry for taking up your time, im suspecting i messed up the drive prior to installation ?

previously i mounted the media```sda2``` in screen shot (which im presuming is just the usb drive contents being``` efi/boot or efi/debian```
and i copied the startup.nsh over but apon restart it didnt do anything unless the next boot sequence is quite long ?

**Reply by alexhorner** [reply]
I see that SDC is isw_raid_member which is the Intel Rapid RAID, likely causing you issues. Like I said I would suggest you do the install onto a single drive rather than attempt to place it onto an array.

Feel free to email me (email on profile) if you'd like to exchange some kind of instant messaging details

**Reply by RedDragon23** [reply]
Hi, I seem to be facing similar issues as @ineptmath. When booting back into the installer, then selecting "Start Shell" which presents a terminal, I try and locate the boot partition, which for me is sdb2 (ext4) I think and then copied over the startup.nsh and its not automatically booted back into Debian (headless, so unable to see the serial output).  I assume the boot partition is the one in the ext4 file system. I also have sdb1(vfat) and sdb 3(swap)
![image_2024-08-05_235346219](https://github.com/user-attachments/assets/3a38ad94-2aa2-43a8-84fd-4811971559e8)

**Reply by alexhorner** [reply]
The boot partition is the vfat partition, which gets mounted to the boot directory within the ext4 partition after boot. You don't want to place the startup.nsh into the boot directory of the ext4 filesystem. You know you got the right partition if there is an EFI directory at the root of the partition.

**Reply by RedDragon23** [reply]
@alexhorner Thank you for the fast response and for clarifying the location. I have managed to copy the startup.nsh to the vFat partition and it has booted into Debian automatically after a reboot! 😄
